plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestDateCoercer.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.type.DateType;
+import io.trino.spi.type.Type;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.time.LocalDate;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.plugin.hive.HiveTimestampPrecision.NANOSECONDS;
+import static io.trino.plugin.hive.HiveType.toHiveType;
+import static io.trino.plugin.hive.coercions.CoercionUtils.createCoercer;
+import static io.trino.spi.predicate.Utils.blockToNativeValue;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.trino.spi.type.VarcharType.createVarcharType;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestDateCoercer
+{
+    @Test(dataProvider = "validDateProvider")
+    public void testValidUnboundedVarcharToDate(String date)
+    {
+        assertVarcharToDateCoercion(createUnboundedVarcharType(), date);
+    }
+
+    @Test(dataProvider = "validDateProvider")
+    public void testValidBounded10VarcharToDate(String date)
+    {
+        assertVarcharToDateCoercion(createVarcharType(10), date);
+    }
+
+    @Test(dataProvider = "validDateProvider")
+    public void testValidBounded7VarcharToDate(String date)
```

Comment:
here we only test coercions and coercer doesn't care about length of input value as long as it contains correct values, in a real word scenario it would fail on INSERT operation, I will just remove these tests as they are missleading